demes/load_dump.py

```diff
@@ -52,6 +54,40 @@ def _dump_yaml_fromdict(data, fp):
         yaml.dump(data)
 
 
+def _stringify_infinities(data: MutableMapping[str, Any]) -> None:
+    """
+    Modifies the data dict so infinite values are set to the string "Infinity".
+
+    This is done for JSON output, because the JSON spec explicitly does not
+    provide an encoding for infinity.  The string-valued "Infinity" is likely
+    to be parsed without problem, and is converted to a float inf by the
+    following implementations:
+     * Python: float("Infinity")
+     * JavaScript: parseFloat("Infinity")
+     * C: strtod("Infinity", NULL) and atof("Infinity")
+     * R: as.numeric("Infinity")
+     * SLiM/Eidos: asFloat("Infinity")
+    """
+    for deme in data["demes"]:
+        if "start_time" in deme and math.isinf(deme["start_time"]):
+            deme["start_time"] = "Infinity"
+    for migration in data.get("migrations", []):
+        if "start_time" in migration and math.isinf(migration["start_time"]):
+            migration["start_time"] = "Infinity"
+
+
+def _unstringify_infinities(data: MutableMapping[str, Any]) -> None:
+    """
+    Modifies the data dict so the string "Infinity" is converted to float.
+    """
+    for deme in data["demes"]:
+        if "start_time" in deme and math.isinf(float(deme["start_time"])):
```

Comment:
This is a bit roundabout isn't it? Why not
```
if deme.get("start_time") == INFINITY:
    deme["start_time"] = math.inf
```
I guess the argument is that yours is more flexible?